### PR TITLE
DataMigration: Adding SQL DB Migration support

### PR DESCRIPTION
**Pull request by ashutoshsuman99**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
DataMigration

File: src/datamigration/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,12 @@
 Release History
 ===============
 
+0.3.0
+++++++
+* [BREAKING CHANGES] Removing parameters provisioing-error and migration-operation-id from commands az datamigration sql-managed-instance create and az datamigration sql-vm create, as they are unnecessary parameters. 
+* [BREAKING CHANGES] Typo correction for parameters --number-of-interation to --number-of-iteration in command az datamigration performance-data-collection
+* [NEW SUBGROUP] az datamigration sql-db - Manage database migrations to SQL DB instance.
```
Comment (evelyn-ys) [inline src/datamigration/HISTORY.rst:10]:
```suggestion
* [BREAKING CHANGE] `az datamigration sql-managed-instance/sql-vm create`: Remove `--provisioing-error` and `--migration-operation-id` as they are unnecessary parameters. 
* [BREAKING CHANGE] `az datamigration performance-data-collection`: Rename `--number-of-interation` to `--number-of-iteration` for typo correction
* [NEW COMMAND GROUP] `az datamigration sql-db`: Manage database migrations to SQL DB instance.
```

File: src/datamigration/azext_datamigration/generated/custom.py [hunk 2]
```diff
@@ -77,11 +152,10 @@ def datamigration_sql_managed_instance_cancel(client,
                                               resource_group_name,
                                               managed_instance_name,
                                               target_db_name,
-                                              migration_operation_id=None,
+                                              migration_operation_id,
                                               no_wait=False):
     parameters = {}
-    if migration_operation_id is not None:
-        parameters['migration_operation_id'] = migration_operation_id
+    parameters['migration_operation_id'] = migration_operation_id
```
Comment (evelyn-ys) [inline src/datamigration/azext_datamigration/generated/custom.py:158]: Is this also a breaking change that `--migration-operation-id` is required now for `az datamigration sql-managed-instance cancel`?

Comment (ashutoshsuman99) [reply src/datamigration/azext_datamigration/generated/custom.py:158]: Yes, actually if necessary to add this parameter even before but not displayed as a required one from CLI help message point view. We have updated that in these changes.  I will add this to history.rst.

File: src/datamigration/azext_datamigration/generated/custom.py [hunk 3]
```diff
@@ -94,11 +168,10 @@ def datamigration_sql_managed_instance_cutover(client,
                                                resource_group_name,
                                                managed_instance_name,
                                                target_db_name,
-                                               migration_operation_id=None,
+                                               migration_operation_id,
                                                no_wait=False):
     parameters = {}
-    if migration_operation_id is not None:
-        parameters['migration_operation_id'] = migration_operation_id
+    parameters['migration_operation_id'] = migration_operation_id
```
[the remark this answers is not in the document]
Comment (ashutoshsuman99) [reply src/datamigration/azext_datamigration/generated/custom.py:174]: Done

File: src/datamigration/azext_datamigration/generated/custom.py (hunk 2 above)
Comment (ashutoshsuman99) [reply src/datamigration/azext_datamigration/generated/custom.py:158]: Done

File: src/datamigration/HISTORY.rst (hunk 1 above)
Comment (ashutoshsuman99) [reply src/datamigration/HISTORY.rst]: Updated.

**Pull request merged**